scotty/pitch_angle_analysis.py

```diff
@@ -0,0 +1,858 @@
+                        "theta_output",
+                        "K_magnitude_array",
+                    ):
+                        current_ds = ds[key]
+                        data = analysis_file[key]
+                        if len(current_ds.dims) < 4:
+                            ds[key] = current_ds.expand_dims(
+                                dim={"trajectory_step": np.arange(0, path_len)}, axis=-1
+                            ).copy()
+                        ds[key].loc[index] = data
+
+                    ds["cutoff_index"].loc[index] = analysis_file["cutoff_index"]
+
+                    analysis_file.close()
+
+                    input_file = np.load(path_input)
+
+                    input_file.close()
+
+                    output_file = np.load(path_output)
+
+                    for key in (
+                        "q_R_array",
+                        "q_zeta_array",
+                        "q_Z_array",
+                        "K_R_array",
+                        "K_Z_array",
+                        "poloidal_flux_output",
+                    ):
+                        current_ds = ds[key]
+                        if len(current_ds.dims) < 4:
+                            ds[key] = current_ds.expand_dims(
+                                dim={"trajectory_step": np.arange(0, path_len)}, axis=-1
+                            ).copy()
+                        ds[key].loc[index] = output_file[key]
+
+                    ds["K_zeta_initial"].loc[index] = output_file["K_zeta_initial"]
+                    output_file.close()
+
+        self.dataset = ds
+        self.dataset["cutoff_index"] = ds["cutoff_index"].astype(int)
+        self.dataset["theta_output"] = np.rad2deg(ds["theta_output"])
+        self.dataset["theta_m_output"] = np.rad2deg(ds["theta_m_output"])
+        self.dataset["delta_theta_m"] = np.rad2deg(ds["delta_theta_m"])
+        self.spline_memo = {}
+
+    #### Class Methods ####
+
+    def set_attrs(self, attribute, value):
+        self.dataset.attrs[attribute] = value
+
+    def set_descriptor(self, descriptor):
+        """Used to set a unique descriptor to identify sweeps with different plasma
+        equilibrium.
+        """
+        self.set_attrs["descriptor"] = descriptor
+
+    ## I/O
+
+    def to_netcdf(self, folder="", filename=None, suffix=""):
+        """Saves contents of Dataset into a netCDF4 file for easy read and writing.
+
+        Args:
+            folder (str, optional): Path to folder to save file to. Default is None.
+            filename (str, optional): Filename to save the nc file as. Default is 'SweepDataset', or
+            the descriptor string if provided.
+        """
+        if not filename:
+            descriptor = self.dataset["descriptor"]
+            if descriptor:
+                filename = descriptor
+            else:
+                filename = "SweepDataset"
+
+        file_path = folder + filename + suffix + ".nc"
```

Comment:
This is fine, but a good habit is to use f-strings for composing strings:

```suggestion
        file_path = f"{folder}{filename}{suffix}.nc"
```

You might also look at [`pathlib`](https://docs.python.org/3/library/pathlib.html) for easier handling of folders and filenames